bundles/framework/layerselector2/view/FilterButtons.js

```diff
@@ -0,0 +1,180 @@
+                    filterIcon.addClass(filter.cls.active);
+                    filterIcon.addClass('active');
+                }
+                // Otherwise use deactive icon
+                else {
+                    filterIcon.addClass(filter.cls.deactive);
+                }
+            });
+        },
+        /**
+         * Activate selected filter.
+         * @method @public activateFilter
+         * @param  {Function} filterName activate filter name
+         */
+        activateFilter: function(filterName) {
+            this._currentFilter = filterName;
+
+            var filterContainer = this.rootElement;
+            var filters = this.layerlistService.getLayerlistFilterButton();
+            Object.keys(filters).forEach(function(key) {
+                var filter = filters[key];
+                var filterIcon = filterContainer.find('.filter-icon.' + 'filter-' + filter.id);
+                if(filter.id === filterName) {
+                    filterIcon.removeClass(filter.cls.deactive);
+                    filterIcon.addClass(filter.cls.active);
+                    filterIcon.addClass('active');
+                } else {
+                    filterIcon.removeClass(filter.cls.active);
+                    filterIcon.addClass(filter.cls.deactive);
+                    filterIcon.removeClass('active');
+                }
+            });
+            this.trigger('FilterActivate', this._currentFilter);
+        },
+        /**
+         * Deactivate all filters
+         * @method  @public deactivateAllFilters
+         *
+         * @param {String} notDeactivateThisFilter not deactivate this filter
+         */
+        deactivateAllFilters: function(notDeactivateThisFilter) {
+            var me = this;
+            me._currentFilter = null;
+            var filterContainer = this.rootElement;
+            var filters = me.layerlistService.getLayerlistFilterButton();
+
+            Object.keys(filters).forEach(function(key) {
+                var filter = filters[key];
+                if (!notDeactivateThisFilter || filter.id !== notDeactivateThisFilter) {
+                    var filterIcon = filterContainer.find('.filter-icon.' + 'filter-' + filter.id);
+                    filterIcon.removeClass(filter.cls.active);
+                    filterIcon.removeClass('active');
+                    filterIcon.addClass(filter.cls.deactive);
+                    // Set tooltip for one per filter
+                        me._setFilterTooltip(filter.name, filter.tooltip);
+                }
+            });
+
+            if (!notDeactivateThisFilter) {
+                me.activateFilter();
+            }
+            this.trigger('FilterActivate', this._currentFilter);
+        },
+        render: function() {
+            var me = this;
+            this.rootElement.empty();
+            this.filterButtons.forEach( function ( button ) {
+                if ( _.include(me.renderableButtons, button.name) ) {
```

Comment:
As this is the only time you need lodash/underscore for the component I would just replace this with:

   if(me.renderableButtons.findIndex(button.name) !== -1) { ... }

Or if renderableButtons is an object with Object.keys() to get rid of the extra dependency.